core/sql/executor/ExExeUtilGet.cpp

```diff
@@ -1818,8 +1813,9 @@ short ExExeUtilGetMetadataInfoTcb::work()
 	    {
                if (!CmpCommon::context()->isAuthorizationEnabled())
                {
-                  ComDiagsArea * diags = getDiagsArea();
-                  *diags << DgSqlCode(-CAT_AUTHORIZATION_NOT_ENABLED);
+                  ComDiagsArea * diagsArea = getDiagsArea();
+                  ExRaiseSqlError(getHeap(), &diagsArea, -CAT_AUTHORIZATION_NOT_ENABLED);
+                  setDiagsArea(diagsArea);
```

Comment:
Having to do these three statements in so many places is not ideal. Would it make sense to make diagsArea_ a protected member, so we could pass it to ExRaiseSqlError here, in just a single statement:

```
ExRaiseSqlError(getHeap(), &diagsArea_, ...);
```

Reply:
Sorry, I'm not sure I understand. What I am suggesting doesn't require you to change any other cases, it just makes this case (and the many repetitions of it in your commit) simpler.

Reply:
I misunderstood your comment. I thought that you wanted to make a function. Yes. I am changing it to pass in diagsArea_ directly, so that we can do this in a single statement. diagsArea_ already has protected mode access.